src/governance/governance.h

```diff
@@ -161,38 +161,38 @@ class CGovernanceManager
     static const int MAX_TIME_FUTURE_DEVIATION;
     static const int RELIABLE_PROPAGATION_TIME;
 
-    int64_t nTimeLastDiff;
+    int64_t nTimeLastDiff GUARDED_BY(cs);
 
     // keep track of current block height
-    int nCachedBlockHeight;
+    int nCachedBlockHeight GUARDED_BY(cs);
 
     // keep track of the scanning errors
-    std::map<uint256, CGovernanceObject> mapObjects;
+    std::map<uint256, CGovernanceObject> mapObjects GUARDED_BY(cs);
 
     // mapErasedGovernanceObjects contains key-value pairs, where
     //   key   - governance object's hash
     //   value - expiration time for deleted objects
-    std::map<uint256, int64_t> mapErasedGovernanceObjects;
+    std::map<uint256, int64_t> mapErasedGovernanceObjects GUARDED_BY(cs);
 
-    std::map<uint256, CGovernanceObject> mapPostponedObjects;
-    hash_s_t setAdditionalRelayObjects;
+    std::map<uint256, CGovernanceObject> mapPostponedObjects GUARDED_BY(cs);
+    hash_s_t setAdditionalRelayObjects GUARDED_BY(cs);
 
-    object_ref_cm_t cmapVoteToObject;
+    object_ref_cm_t cmapVoteToObject GUARDED_BY(cs) {MAX_CACHE_SIZE};
 
-    CacheMap<uint256, CGovernanceVote> cmapInvalidVotes;
+    CacheMap<uint256, CGovernanceVote> cmapInvalidVotes GUARDED_BY(cs) {MAX_CACHE_SIZE};
 
-    vote_cmm_t cmmapOrphanVotes;
+    vote_cmm_t cmmapOrphanVotes GUARDED_BY(cs) {MAX_CACHE_SIZE};
 
-    txout_m_t mapLastMasternodeObject;
+    txout_m_t mapLastMasternodeObject GUARDED_BY(cs);
 
-    hash_s_t setRequestedObjects;
+    hash_s_t setRequestedObjects GUARDED_BY(cs);
 
-    hash_s_t setRequestedVotes;
+    hash_s_t setRequestedVotes GUARDED_BY(cs);
 
-    bool fRateChecksEnabled;
+    bool fRateChecksEnabled GUARDED_BY(cs) {true};
 
     // used to check for changed voting keys
-    CDeterministicMNListPtr lastMNListForVotingKeys;
+    CDeterministicMNListPtr lastMNListForVotingKeys GUARDED_BY(cs);
```

Comment:
it should be 
CDeterministicMNListPtr lastMNListForVotingKeys GUARDED_BY(cs) {std::make_shared<CDeterministicMNList>()};

Also add below header file as forward declaration is not sufficient
#include <evo/deterministicmns.h>